app/__init__.py

```diff
@@ -33,6 +36,37 @@ def validate_origin():
         abort(403, 'Permission denied')
 
 
+@app.after_request
+def add_charset(response):
+    # Python uses UTF-8 as charset by default
+    if response.headers.get('Content-Type') == 'application/json':
+        response.headers.set('Content-Type', 'application/json; charset=utf-8')
+    return response
+
+
+# Add CORS Headers to all request
+@app.after_request
+def add_cors_header(response):
+    if (
+        'Origin' in request.headers and
+        re.match(ALLOWED_DOMAINS_PATTERN, request.headers['Origin'])
+    ):
+        response.headers.set('Access-Control-Allow-Origin', request.headers['Origin'])
+        response.headers.set('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, HEAD')
```

Comment:
Will all methods be returned in the answer, even if the OPTIONS request is on a GET endpoint?

Reply:
Probably I misunderstood this example: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS#preflighted_requests_in_cors

But if there was a preflighted request to e.g. /<shortlink-id>, which is a GET endpoint, would it then make sense to have `POST` in the response's `Access-Control-Allow-Methods`?

Reply:
Good point, I'll change that.

Reply:
Thanks. Could also be an issue in other flask services, I guess?

Reply:
Yes, but try not being too smart, because for every "improvement" you make on the services, I'll have to fix outdated and legacy technology.

Reply:
@procrastinatio could you test it with sys-s.dev.bgdi.ch does it works with angular ?